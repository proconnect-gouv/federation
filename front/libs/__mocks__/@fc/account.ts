import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
