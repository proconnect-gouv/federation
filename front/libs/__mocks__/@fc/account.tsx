import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
