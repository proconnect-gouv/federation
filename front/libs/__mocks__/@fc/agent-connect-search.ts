import * as React from 'react';

export const AgentConnectSearchContext = React.createContext({
  hasSearched: false,
  payload: {
    csrfToken: 'csrf-token-mock',
    identityProviders: [],
    isLoaded: false,
    ministries: [],
    serviceProviderName: undefined,
  },
  searchResults: [],
  setSearchTerm: () => {},
});

export const useSearchResults = jest.fn(() => ({
  searchResults: [
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.1',
          uid: 'mock-idp-uid-1.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.2',
          uid: 'mock-idp-uid-1.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-1',
        identityProviders: ['mock-idp-uid-1'],
        name: 'mock-ministry-name-1',
        sort: 1,
      },
    },
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.1',
          uid: 'mock-idp-uid-2.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.2',
          uid: 'mock-idp-uid-2.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-2',
        identityProviders: ['mock-idp-uid-2'],
        name: 'mock-ministry-name-2',
        sort: 0,
      },
    },
  ],
  showNoResults: false,
  showResults: true,
}));

export const useOnSearch = jest.fn(() => jest.fn());
