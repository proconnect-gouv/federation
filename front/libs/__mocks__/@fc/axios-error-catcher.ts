import React from 'react';

export const AxiosErrorCatcherProvider = jest.fn();

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));
