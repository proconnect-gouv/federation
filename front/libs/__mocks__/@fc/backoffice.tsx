export const FieldCheckboxComponent = jest.fn(() => <div>FieldCheckboxComponent</div>);

export const FieldSwitchComponent = jest.fn(() => <div>FieldSwitchComponent</div>);

export const ButtonSimpleComponent = jest.fn(() => <div>ButtonSimpleComponent</div>);
