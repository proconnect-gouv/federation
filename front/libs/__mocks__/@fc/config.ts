export const ConfigService = {
  get: jest.fn(),
  initialize: jest.fn(),
};
