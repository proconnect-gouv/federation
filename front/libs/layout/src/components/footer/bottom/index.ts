/* istanbul ignore file */

// declarative file
export * from './layout-footer.bottom';
