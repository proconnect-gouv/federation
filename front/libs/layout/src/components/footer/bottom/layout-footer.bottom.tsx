import React from 'react';

import { ConfigService } from '@fc/config';

import { Options } from '../../../enums';
import type { LayoutConfig } from '../../../interfaces';

export const LayoutFooterBottomComponent = React.memo(() => {
  const config = ConfigService.get<LayoutConfig>(Options.CONFIG_NAME);
  const { navigation } = config.footer;
  const { showLicence } = config.features;

  return (
    <div className="fr-footer__bottom" data-testid="sticky-footer-fr-footer__bottom">
      {navigation && (
        <ul className="fr-footer__bottom-list">
          {navigation.map(({ href, label, title }, index) => {
            const uniqKey = `layout-footer-bottom-links::${index}`;
            return (
              <li key={uniqKey} className="fr-footer__bottom-item">
                <a className="fr-footer__bottom-link" href={href} title={title}>
                  {label}
                </a>
              </li>
            );
          })}
        </ul>
      )}
      {showLicence && (
        <div className="fr-footer__bottom-copy">
          <p>
            Sauf mention contraire, tous les contenus de ce site sont sous{' '}
            <a
              href="https://github.com/etalab/licence-ouverte/blob/master/LO.md"
              rel="noreferrer"
              target="_blank">
              licence etalab-2.0
            </a>
          </p>
        </div>
      )}
    </div>
  );
});

LayoutFooterBottomComponent.displayName = 'LayoutFooterBottomComponent';
