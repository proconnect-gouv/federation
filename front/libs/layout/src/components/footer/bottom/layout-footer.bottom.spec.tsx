import { render } from '@testing-library/react';

import { ConfigService } from '@fc/config';

import { LayoutFooterBottomComponent } from './layout-footer.bottom';

describe('LayoutFooterBottomComponent', () => {
  // given
  const navigationMock = [
    {
      href: 'any-href-mock-1',
      label: 'any-label-mock-1',
      title: 'any-title-mock-1',
    },
    {
      href: 'any-href-mock-2',
      label: 'any-label-mock-2',
      title: 'any-title-mock-2',
    },
  ];

  beforeEach(() => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      features: { showLicence: expect.any(Boolean) },
      footer: { navigation: [expect.any(Object), expect.any(Object)] },
    });
  });

  it('should call ConfigService.get with Layout config name', () => {
    // when
    render(<LayoutFooterBottomComponent />);

    // then
    expect(ConfigService.get).toHaveBeenCalledOnce();
    expect(ConfigService.get).toHaveBeenCalledWith('Layout');
  });

  it('should match the snapshot when features.showLicence is true and footer.navigation is defined', () => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      features: { showLicence: true },
      footer: { navigation: navigationMock },
    });

    // when
    const { container } = render(<LayoutFooterBottomComponent />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should render the navigation when footer.navigation is defined', async () => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      features: { showLicence: false },
      footer: { navigation: navigationMock },
    });

    // when
    const { container, findAllByRole, getByTitle } = render(<LayoutFooterBottomComponent />);
    const elementLinks = await findAllByRole('link');
    const elementLink1 = getByTitle('any-title-mock-1');
    const elementLink2 = getByTitle('any-title-mock-2');

    // then
    expect(container).toMatchSnapshot();
    expect(elementLinks).toHaveLength(2);
    expect(elementLink1).toBeInTheDocument();
    expect(elementLink1).toHaveAttribute('href', 'any-href-mock-1');
    expect(elementLink1).toHaveAttribute('title', 'any-title-mock-1');
    expect(elementLink1).toHaveAttribute('title', 'any-title-mock-1');
    expect(elementLink1).toHaveTextContent('any-label-mock-1');
    expect(elementLink2).toBeInTheDocument();
    expect(elementLink2).toHaveAttribute('href', 'any-href-mock-2');
    expect(elementLink2).toHaveAttribute('title', 'any-title-mock-2');
    expect(elementLink2).toHaveAttribute('title', 'any-title-mock-2');
    expect(elementLink2).toHaveTextContent('any-label-mock-2');
  });

  it('should render the licence when features.showLicence is defined', () => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      features: { showLicence: true },
      footer: { navigation: [] },
    });

    // when
    const { container, getByRole } = render(<LayoutFooterBottomComponent />);
    const elementLink = getByRole('link');

    // then
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(
      'Sauf mention contraire, tous les contenus de ce site sont sous licence etalab-2.0',
    );
    expect(elementLink).toHaveAttribute(
      'href',
      'https://github.com/etalab/licence-ouverte/blob/master/LO.md',
    );
  });
});
