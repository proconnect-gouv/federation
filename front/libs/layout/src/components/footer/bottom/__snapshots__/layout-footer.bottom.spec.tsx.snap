// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutFooterBottomComponent should match the snapshot when features.showLicence is true and footer.navigation is defined 1`] = `
<div>
  <div
    class="fr-footer__bottom"
    data-testid="sticky-footer-fr-footer__bottom"
  >
    <ul
      class="fr-footer__bottom-list"
    >
      <li
        class="fr-footer__bottom-item"
      >
        <a
          class="fr-footer__bottom-link"
          href="any-href-mock-1"
          title="any-title-mock-1"
        >
          any-label-mock-1
        </a>
      </li>
      <li
        class="fr-footer__bottom-item"
      >
        <a
          class="fr-footer__bottom-link"
          href="any-href-mock-2"
          title="any-title-mock-2"
        >
          any-label-mock-2
        </a>
      </li>
    </ul>
    <div
      class="fr-footer__bottom-copy"
    >
      <p>
        Sauf mention contraire, tous les contenus de ce site sont sous
         
        <a
          href="https://github.com/etalab/licence-ouverte/blob/master/LO.md"
          rel="noreferrer"
          target="_blank"
        >
          licence etalab-2.0
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`LayoutFooterBottomComponent should render the licence when features.showLicence is defined 1`] = `
<div>
  <div
    class="fr-footer__bottom"
    data-testid="sticky-footer-fr-footer__bottom"
  >
    <ul
      class="fr-footer__bottom-list"
    />
    <div
      class="fr-footer__bottom-copy"
    >
      <p>
        Sauf mention contraire, tous les contenus de ce site sont sous
         
        <a
          href="https://github.com/etalab/licence-ouverte/blob/master/LO.md"
          rel="noreferrer"
          target="_blank"
        >
          licence etalab-2.0
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`LayoutFooterBottomComponent should render the navigation when footer.navigation is defined 1`] = `
<div>
  <div
    class="fr-footer__bottom"
    data-testid="sticky-footer-fr-footer__bottom"
  >
    <ul
      class="fr-footer__bottom-list"
    >
      <li
        class="fr-footer__bottom-item"
      >
        <a
          class="fr-footer__bottom-link"
          href="any-href-mock-1"
          title="any-title-mock-1"
        >
          any-label-mock-1
        </a>
      </li>
      <li
        class="fr-footer__bottom-item"
      >
        <a
          class="fr-footer__bottom-link"
          href="any-href-mock-2"
          title="any-title-mock-2"
        >
          any-label-mock-2
        </a>
      </li>
    </ul>
  </div>
</div>
`;
